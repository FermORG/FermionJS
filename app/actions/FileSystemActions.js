// @flow
export const SET_ACTIVE_COMPONENT = 'SET_ACTIVE_COMPONENT';

export function setActiveComponent(component: string) {
  return {
    type: SET_ACTIVE_COMPONENT,
    component,
  };
}
