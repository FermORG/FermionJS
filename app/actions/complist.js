//@flow
//complist actions.


// import type { addComponentsStateType } from '../reducers/complist';

// type actionType = {
//   +type: string,
// };

export const ADD_COMPONENTS = 'ADD_COMPONENTS';

export const ADD_TO_PROJECT = 'ADD_TO_PROJECT';

export function AddComponents(components) {
  return {
    type: 'ADD_COMPONENTS',
    components
  };
}

export function AddToProject(component) {
  return {
    type: 'ADD_TO_PROJECT',
    component
  };
}
