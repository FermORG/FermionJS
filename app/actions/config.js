// @flow
// config actions
export const ADD_STATE = 'ADD_STATE';
export const ADD_PROPS = 'ADD_PROPS';
export const ADD_STYLES = 'ADD_STYLES';
export const ADD_EVENTS = 'ADD_EVENTS';
export const DELETE_STATE = 'DELETE_STATE';
export const DELETE_PROPS = 'DELETE_PROPS';
export const DELETE_STYLES = 'DELETE_STYLES';
export const DELETE_EVENTS = 'DELETE_EVENTS';

export function addState(aState, component) {
  return {
    type: 'ADD_STATE',
    aState,
  };
}

export function addProps(prop, component) {
  return {
    type: 'ADD_PROPS',
    prop,
    component
  };
}

export function addStyles(style, component) {
  return {
    type: 'ADD_STYLES',
    style,
    component
  };
}

export function addEvents(event, component) {
  return {
    type: 'ADD_EVENTS',
    event,
    component
  };
}

export function deleteState(stateKey, component) {
  return {
    type: 'DELETE_STATE',
    propKey: stateKey,
  };
}
export function deleteProps(propKey, component) {
  return {
    type: 'DELETE_PROPS',
    prop: propKey,
    component
  };
}
export function deleteStyles(styleKey, component) {
  return {
    type: 'DELETE_STYLES',
    style: styleKey,
    component
  };
}
export function deleteEvents(eventKey, component) {
  return {
    type: 'DELETE_EVENTS',
    event: eventKey,
    component
  };
}
