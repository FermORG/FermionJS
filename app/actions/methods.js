export const METHODS = 'METHODS';


export function updateMethods(methods) {
  return {
    type: METHODS,
    methods
  };
}
