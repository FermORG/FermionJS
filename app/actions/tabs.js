// @flow

// tabs actions.

import type { activeTabStateType } from '../reducers/tabs';

type actionType = {
  +type: string
};

export const TOGGLE_STATE = 'TOGGLE_STATE';
export const TOGGLE_PROPS = 'TOGGLE_PROPS';
export const TOGGLE_EVENTS = 'TOGGLE_EVENTS';
export const TOGGLE_STYLES = 'TOGGLE_STYLES';
export const TOGGLE_TAB = 'TOGGLE_TAB';

export function toggleTabs(name: string) {
  switch (name) {
    case 'Props':
      return {
        type: TOGGLE_PROPS,
        name,
      };
    case 'Events':
      return {
        type: TOGGLE_EVENTS,
        name,
      };
    case 'Styles':
      return {
        type: TOGGLE_STYLES,
        name,
      };
    default:
      return {
        type: TOGGLE_STATE,
        name,
      };
  }
}
