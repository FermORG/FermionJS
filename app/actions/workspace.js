import { WORKSPACE_ID } from './../constants';

export const ADD_CHILD = 'ADD_CHILD';
export const CREATE_CHILD = 'CREATE_CHILD';
export const DELETE_CHILD = 'DELETE_CHILD';
export const MOVE_CHILD = 'MOVE_CHILD';
export const REMOVE_CHILD = 'REMOVE_CHILD';
export const UPDATE_STYLE = 'UPDATE_STYLE';

const createChild = ({ name, props }) => ({
  name,
  props,
  children: [],
  parentID: WORKSPACE_ID,
});

// export function addChild(targetID, newComponentName) {
//   const newComponent = createChild(newComponentName);
export function addChild(targetID, component) {
  const newComponent = createChild(component);
  return {
    type: ADD_CHILD,
    newComponent,
    targetID
  };
}

export function removeChild(id) {
  return {
    type: REMOVE_CHILD,
    id
  };
}

export function moveChild(sourceID, targetID) {
  return {
    type: MOVE_CHILD,
    targetID,
    sourceID
  };
}

export function updateStyle(sourceID, newStyle) {
  return {
    type: UPDATE_STYLE,
    sourceID,
    newStyle
  };
}
