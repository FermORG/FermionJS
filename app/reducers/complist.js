// @flow
// complist reducers
import { ADD_COMPONENTS, ADD_TO_PROJECT } from '../actions/complist';

type actionType = {
  +type: string,
};

const defaultData = {
  availableComponents: [
    {
      name: 'One',
      styles: {
        backgroundColor: 'red',
        height: '50%',
        width: '50%'
      }
    },
    {
      name: 'Two',
      styles: {
        backgroundColor: 'yellow',
        height: '50%',
        width: '50%'
      }
    },
    {
      name: 'Five',
      styles: {
        backgroundColor: 'white',
        height: '50%',
        width: '50%'
      },
    },
    {
      name: 'Four',
      styles: {
        backgroundColor: 'blue',
        height: '50%',
        width: '50%'
      },
    }
  ]
};

export default function listReducer(state: {} = defaultData, action: actionType) {
  const newState = Object.assign({}, state);
  // newState.availableComponents =JSON.parse(JSON.stringify(newState.availableComponents));
  switch (action.type) {
    case ADD_COMPONENTS:
      return newState.availableComponents.concat(action.components);
    case ADD_TO_PROJECT:
      return newState;
    default:
      return newState;
  }
}
