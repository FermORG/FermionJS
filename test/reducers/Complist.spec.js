import { ADD_COMPONENTS, ADD_TO_PROJECT } from '../../app/actions/complist';
import listReducer from '../../app/reducers/complist';

const testData = {
  components: [
    {
      name: 'Six',
      styles: {
        backgroundColor: 'white',
        height: '50%',
        width: '50%'
      },
    },
    {
      name: 'Seven',
      styles: {
        backgroundColor: 'blue',
        height: '50%',
        width: '50%'
      },
    },
  ]
};

const defaultData = {
  availableComponents: [
    {
      name: 'One',
      styles: {
        backgroundColor: 'red',
        height: '50%',
        width: '50%'
      }
    },
    {
      name: 'Two',
      styles: {
        backgroundColor: 'yellow',
        height: '50%',
        width: '50%'
      }
    },
    {
      name: 'Five',
      styles: {
        backgroundColor: 'white',
        height: '50%',
        width: '50%'
      },
    },
    {
      name: 'Four',
      styles: {
        backgroundColor: 'blue',
        height: '50%',
        width: '50%'
      },
    }
  ]
};


describe('component listReducer', () => {
  it('should add testData to the components list', () => {
    expect(listReducer(defaultData, {
      type: ADD_COMPONENTS,
      components: testData.components,
    })).toMatchSnapshot();
  });

  it('should return default data when presented with gibberish', () => {
    expect(listReducer(defaultData, {
      type: 'GIBBERISH',
      components: testData.components,
    })).toMatchSnapshot();
  });

  it('should return default data when the action type is ADD_TO_PROJECT', () => {
    expect(listReducer(defaultData, {
      type: ADD_TO_PROJECT,
      components: testData.components,
    })).toMatchSnapshot();
  });

});
