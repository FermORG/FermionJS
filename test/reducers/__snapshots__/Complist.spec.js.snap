// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component listReducer should add testData to the components list 1`] = `
Array [
  Object {
    "name": "One",
    "styles": Object {
      "backgroundColor": "red",
      "height": "50%",
      "width": "50%",
    },
  },
  Object {
    "name": "Two",
    "styles": Object {
      "backgroundColor": "yellow",
      "height": "50%",
      "width": "50%",
    },
  },
  Object {
    "name": "Five",
    "styles": Object {
      "backgroundColor": "white",
      "height": "50%",
      "width": "50%",
    },
  },
  Object {
    "name": "Four",
    "styles": Object {
      "backgroundColor": "blue",
      "height": "50%",
      "width": "50%",
    },
  },
  Object {
    "name": "Six",
    "styles": Object {
      "backgroundColor": "white",
      "height": "50%",
      "width": "50%",
    },
  },
  Object {
    "name": "Seven",
    "styles": Object {
      "backgroundColor": "blue",
      "height": "50%",
      "width": "50%",
    },
  },
]
`;

exports[`component listReducer should return default data when presented with gibberish 1`] = `
Object {
  "availableComponents": Array [
    Object {
      "name": "One",
      "styles": Object {
        "backgroundColor": "red",
        "height": "50%",
        "width": "50%",
      },
    },
    Object {
      "name": "Two",
      "styles": Object {
        "backgroundColor": "yellow",
        "height": "50%",
        "width": "50%",
      },
    },
    Object {
      "name": "Five",
      "styles": Object {
        "backgroundColor": "white",
        "height": "50%",
        "width": "50%",
      },
    },
    Object {
      "name": "Four",
      "styles": Object {
        "backgroundColor": "blue",
        "height": "50%",
        "width": "50%",
      },
    },
  ],
}
`;

exports[`component listReducer should return default data when the action type is ADD_TO_PROJECT 1`] = `
Object {
  "availableComponents": Array [
    Object {
      "name": "One",
      "styles": Object {
        "backgroundColor": "red",
        "height": "50%",
        "width": "50%",
      },
    },
    Object {
      "name": "Two",
      "styles": Object {
        "backgroundColor": "yellow",
        "height": "50%",
        "width": "50%",
      },
    },
    Object {
      "name": "Five",
      "styles": Object {
        "backgroundColor": "white",
        "height": "50%",
        "width": "50%",
      },
    },
    Object {
      "name": "Four",
      "styles": Object {
        "backgroundColor": "blue",
        "height": "50%",
        "width": "50%",
      },
    },
  ],
}
`;
